src/Core/Form/IdentifiableObject/DataHandler/CarrierFormDataHandler.php

```diff
@@ -126,6 +126,10 @@ private function formatFormRangesData(array $data): array
 
         foreach ($data as $zone) {
             foreach ($zone['ranges'] as $range) {
+                // Due to this bug https://github.com/PrestaShop/PrestaShop/issues/37151,
+                // We have to do this quick and dirty fix.
+                // This bug is due to Symfony request handle that does not remove an empty row
+            //  // even when we use delete_empty (@todo investigate how this could be improved on the form type configuration level).
```

Comment:
```suggestion
```
We just need to add 'allow_delete' => true in the src/PrestaShopBundle/Form/Admin/Improve/Shipping/Carrier/Type/CostsZoneType.php ranges field